### DISCUSSION TITLE
Server-side secrets with Vercel

### DISCUSSION THREAD

**Discussion by jasonsilberman** [vercel/next.js#12370] [Help]
I am building a Next.js site with API Routes and deploying it on Vercel. I am trying to access environment variables on the server within the API Routes.

Trying to access the environment variables I have in `.env` with `process.env.NAME` does not seem to be working when running locally using `next`.

My next thought was to use [Runtime Configuration](https://nextjs.org/docs/api-reference/next.config.js/runtime-configuration) and setup `serverRuntimeConfig` using through `next.config.js`, however on that page is states:

> Runtime configuration is not available when using the serverless target.

Which AFAIK is used when applying to Vercel. So my question is, what is the best way to store manage secrets for API Routes so that I can use access them locally and as well as when running Vercel?

I would like to be able to use a `.env` file locally and then use Vercel's built in Environment Variable support when deploying.

Thanks!

**Comment by jasonsilberman** [comment]
I have found the following solution to my issue, however I want to make sure that I am not leaking secrets into the client side bundle:

I have included the following lines in my `next.config.js`:
```js
const dev = process.env.NODE_ENV !== "production";
if (dev) require("dotenv").config();
```

Will this include the environment variables within the bundle? My guess is that it will not but I want to double check.

**Comment by timneutkens** [comment]
You'll likely want to use this when it's released: https://github.com/zeit/next.js/discussions/11106
